### DISCUSSION TITLE
QR with pivoting for tall and skinny matrices

### DISCUSSION THREAD

**Discussion by langou** [Reference-LAPACK/lapack#668] [Ideas]
For tall and skinny matrices, say 100,000-by-1,000, instead of calling GEQP3 on the tall and skinny matrix, it might be better to call GEQRF on the tall and skinny, and then GEQP3 on the 1,000-by-1,000. We can have a branch in GEQP3 to do this. Based on a ratio for m/n. This would be akin to what we do for SVD. The main difficulty is compatibility of interface with GEPQ3. It would be easier for the GEQRF+GEQP3 algorithm to return Q and R, as opposed to ``Householder vectors`` and R.

**Comment by rileyjmurray** [comment]
@langou, when you say "return Q and R," do you mean return the explicit representation of the economy Q?

**Reply by langou** [reply]
yes
